### ISSUE TITLE
Move the content from exarkun's Twisted Web in 60 Seconds series into the Twisted Web howto

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4192]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#4192|
|Type|task|
|Created|2009-12-27 16:50:15Z|
|Branch|https://github.com/twisted/twisted/tree/web-in60-4192|

Index for the content is here: http://jcalderone.livejournal.com/50562.html

<details><summary>Searchable metadata</summary>

```
trac-id__4192 4192
type__task task
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__branches_web_in60_4192 branches/web-in60-4192
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1261932615000000 1261932615000000
changetime__1266704221000000 1266704221000000
version__None None
owner__ 
cc__jesstess cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess commented|
|-|-|

(In [27894]) Branching to 'web-in60-4192'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Are you going to make an additional document, or cramp it all in the existing howto (which is probably already long enough)?

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess removed owner|
|-|-|

Example organization of the twisted web in 60 seconds series with the first example: serving static content. The web howto has a link to the 60 seconds index which has links to all the examples, so there's minimal additional clutter on the howto index. 

There are up to 17 existing examples that could go into this series. One thing that I know is missing is an attribution to exarkun.

Comments on the organization before I put in work on the rest of the examples?

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|TimAllen set owner to TimAllen|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|TimAllen set owner to @jesstess|
|-|-|

1.It seems to me that if exarkun's series goes on long enough, it could _replace_ the other Web documentation, rather than accompanying it. In particular, it seems a shame to say "This set of examples is intended to be a supplement to the Using Twisted Web tutorial", when it's so much more readable than that tutorial. Maybe you could say something like "for subjects not covered here, see the Twisted Web tutorial and the API documentation".
 1. You might want to add `class="python"` to all the `<pre/>` tags that contain Python source, not just the block of code at the end.

Apart from that, looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess removed owner|
|-|-|

TimAllen's review comments are addressed, and parts 1-6 are now in the branch. These parts are:
 * Serving static content from a directory
 * Generating a page dynamically
 * Static URL dispatch
 * Dynamic URL dispatch
 * Error handling
 * Custom response codes

A double check on the API links would be most appreciated as part of a review. After that I'll add the middle third of the examples.

**Comment by twisted-trac** [comment]
|<img alt="gas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|gas commented|
|-|-|

link seemed fine to me when tested via firefox. Only issue i had was firefox didnt seem to take nay notice of the <pre class="python" tags and interpreted the html tags in the render_GET messages ie output on browser

    def render_GET(self, request):
        return "

%s

" % (calendar(self.year),)

I presume this is a browser xhmtl parsing failure.

Didnt see any attribution to exarkun - maybe add link to original and attribution in index?

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess removed owner|
|-|-|

Thanks for the review, gas!

 * gas is working out lore issues, hence the above review comment about the xhtml looking fun.
 * exarkun said he'd prefer no attribution.

Back up for review!

**Comment by twisted-trac** [comment]
|<img alt="gas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_11"></a>|gas commented|
|-|-|

ok reviewed with lore all looks fine to me. maybe a suggestion for all doc should we put a last updated on them - so if newbies reading it can see if it is out of date?

**Comment by twisted-trac** [comment]
|<img alt="gas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|gas set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_13"></a>|@jesstess commented|
|-|-|

Parts 7-12 have now been added to the branch. These parts are:
 * handling POSTs
 * rpy scripts (or, how to save yourself some typing)
 * asynchronous responses
 * asynchronous responses (via Deferred)
 * interrupted responses
 * logging errors

As before, a double check on the API links would be most appreciated as part of a review. After that I'll add the final third of the examples.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_15"></a>|@therve set owner to @jesstess|
|-|-|

The reference to notifyFinish is wrong in interrupted.xhtml. Everything else looks great.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_15"></a>|@jesstess commented|
|-|-|

(In [28251]) Correct notifyFinish link in interrupted.xhtml

refs [#4192](https://github.com/twisted/twisted/issues/4192)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_16"></a>|@jesstess removed owner|
|-|-|

Parts 13-17 have now been added to the branch. These parts are:

 * WSGIs
 * HTTP authentication
 * Session basics
 * Storing objects in the session
 * Session endings

As before, a double check on the API links would be most appreciated as part of a review. This is the final set to review before inclusion!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_17"></a>|@therve set owner to @jesstess|
|-|-|

Great work! Just one comment: the first Guard link in http-auth is broken.

+1!

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_18"></a>|@jesstess commented|
|-|-|

(In [28284]) Fix guard link.

refs [#4192](https://github.com/twisted/twisted/issues/4192)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_19"></a>|@jesstess set status to `closed`|
|-|-|

(In [28287]) Merge web-in60-4192

Author: jesstess
Reviewers: TimAllen, gas, therve
Fixes: [#4192](https://github.com/twisted/twisted/issues/4192)

Twisted Web in 60 Seconds, a series of short tutorials with
self-contained examples on a range of common web topics, is now a part
of the Twisted Web howto documentation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
